src/main/java/com/google/cloud/spanner/pgadapter/ConnectionHandler.java

```diff
@@ -167,8 +167,10 @@ private void handleError(DataOutputStream output, Exception e) throws Exception
 
   /** Closes portals and statements if the result of an execute was the end of a transaction. */
   public void cleanUp(IntermediateStatement statement) throws Exception {
-    if (!statement.isHasMoreData() && statement.isBound()) {
-      statement.close();
+    for (int index = 0; index < statement.getStatementCount(); index++) {
+      if (!statement.isHasMoreData(index) && statement.isBound()) {
+        statement.close();
```

Comment:
Seems like a mistake. I think that in some version of the code `statement.close()` took an index as an argument, or otherwise this was added as I was planning on adding index as an argument to `statement.close()`. But your current observation is right: This does not make sense in this way (but do note the `statement.hasMoreData(index)` call; it might very well be that in a final version something like this will be needed to close all resources that were opened by the batch)